docs/kap_proposals/kap_11_hashicorp_vault_transit.md

```diff
@@ -0,0 +1,151 @@
+# Hashicorp Vault Transit
+
+This feature allows the user to fetch secrets from [Hashicorp Vault](https://www.vaultproject.io/), with the new secret backend keyword 'vaulttransit'.
+
+Author: [@xqp](https://github.com/xqp) [@Moep90](https://github.com/Moep90)
+## Specification
+
+The following variables need to be exported to the environment(depending on authentication used) where you will run `kapitan refs --reveal` in order to authenticate to your HashiCorp Vault instance:
+* VAULT_ADDR: URL for vault
+* VAULT_SKIP_VERIFY=true: if set, do not verify presented TLS certificate before communicating with Vault server. Setting this variable is not recommended except during testing
+* VAULT_TOKEN: token for vault or file (~/.vault-tokens)
+* VAULT_ROLE_ID: required by approle
+* VAULT_SECRET_ID: required by approle
+* VAULT_USERNAME: username to login to vault
+* VAULT_PASSWORD: password to login to vault
+* VAULT_CLIENT_KEY: the path to an unencrypted PEM-encoded private key matching the client certificate
+* VAULT_CLIENT_CERT: the path to a PEM-encoded client certificate for TLS authentication to the Vault server
+* VAULT_CACERT: the path to a PEM-encoded CA cert file to use to verify the Vault server TLS certificate
+* VAULT_CAPATH: the path to a directory of PEM-encoded CA cert files to verify the Vault server TLS certificate
+* VAULT_NAMESPACE: specify the Vault Namespace, if you have one
+
+Considering any stringdata like `any.value:whatever-you_may*like` ( in our case let’s encrypt `any.value:whatever-you_may*like` with vault transit ) using the key `2022-02-13-test` in a transit secret engine with mount `mytransit` on the vault server, to use this as a secret either follow:
+
+```shell
+$ echo "any.value:whatever-you_may*like" > somefile.txt
+$  kapitan refs --write vaulttransit:<target_name>/to/secret_inside_kapitan --file somefile.txt --target <target_name>
+```
+or in a single line
+```shell
+$ echo "any.value:whatever-you_may*like"  | kapitan refs --write vaulttransit:<target_name>/to/secret_inside_kapitan -t <target_name> -f -
+```
+The entire string __"any.value:whatever-you_may*like"__ will be encrypted by vault and looks like this in return: `vault:v2:Jhn3UzthKcJ2s+sEiO60EUiDmuzqUC4mMBWp2Vjg/DGl+GDFEDIPmAQpc5BdIefkplb6yrJZq63xQ9s=`. This then gets base64 encoded and stored in the secret_inside_kapitan. Now secret_inside_kapitan contains the following
+
+```yaml
+data: dmF1bHQ6djI6SmhuM1V6dGhLY0oycytzRWlPNjBFVWlEbXV6cVVDNG1NQldwMlZqZy9ER2wrR0RGRURJUG1BUXBjNUJkSWVma3BsYjZ5ckpacTYzeFE5cz0=
+encoding: original
+type: vaulttransit
+vault_params:
+  VAULT_ADDR: http://127.0.0.1:8200
+  VAULT_SKIP_VERIFY: 'True'
+  VAULT_TOKEN: s.i53a1DL83REM61UxlJKLdQDY
```

Comment:
This only represents the input of `vault_params` within `parameters.kapitan.secrets.vaulttransit`.
It will not be dumped if its specified via `ENV` like `export VAULT_TOKEN=xyz`
Same applies for vaultkv [here](https://github.com/kapicorp/kapitan/blob/826f618800e1db37700559e797567c90fa0db42f/kapitan/refs/secrets/vaultkv.py#L221)